examples/AccountManagement/GetAccountHierarchy.php

```diff
@@ -0,0 +1,196 @@
+ * limitations under the License.
+ */
+
+namespace Google\Ads\GoogleAds\Examples\AccountManagement;
+
+require __DIR__ . '/../../vendor/autoload.php';
+
+use GetOpt\GetOpt;
+use Google\Ads\GoogleAds\Examples\Utils\ArgumentNames;
+use Google\Ads\GoogleAds\Examples\Utils\ArgumentParser;
+use Google\Ads\GoogleAds\Lib\OAuth2TokenBuilder;
+use Google\Ads\GoogleAds\Lib\V2\GoogleAdsClient;
+use Google\Ads\GoogleAds\Lib\V2\GoogleAdsClientBuilder;
+use Google\Ads\GoogleAds\Lib\V2\GoogleAdsException;
+use Google\Ads\GoogleAds\V2\Errors\GoogleAdsError;
+use Google\Ads\GoogleAds\V2\Resources\CustomerClient;
+use Google\Ads\GoogleAds\V2\Services\GoogleAdsRow;
+use Google\ApiCore\ApiException;
+
+/**
+ * This example gets the account hierarchy of the specified customer account.
+ */
+class GetAccountHierarchy
+{
+    const CUSTOMER_ID = 'INSERT_CUSTOMER_ID_HERE';
+    const PAGE_SIZE = 1000;
+
+    public static function main()
+    {
+        // Either pass the required parameters for this example on the command line, or insert them
+        // into the constants above.
+        $options = (new ArgumentParser())->parseCommandArguments([
+            ArgumentNames::CUSTOMER_ID => GetOpt::REQUIRED_ARGUMENT
+        ]);
+
+        // Generate a refreshable OAuth2 credential for authentication.
+        $oAuth2Credential = (new OAuth2TokenBuilder())->fromFile()->build();
+
+        // Construct a Google Ads client configured from a properties file and the
+        // OAuth2 credentials above.
+        $googleAdsClient = (new GoogleAdsClientBuilder())->fromFile()
+            ->withOAuth2Credential($oAuth2Credential)
+            ->build();
+
+        try {
+            self::runExample(
+                $googleAdsClient,
+                $options[ArgumentNames::CUSTOMER_ID] ?: self::CUSTOMER_ID
+            );
+        } catch (GoogleAdsException $googleAdsException) {
+            printf(
+                "Request with ID '%s' has failed.%sGoogle Ads failure details:%s",
+                $googleAdsException->getRequestId(),
+                PHP_EOL,
+                PHP_EOL
+            );
+            foreach ($googleAdsException->getGoogleAdsFailure()->getErrors() as $error) {
+                /** @var GoogleAdsError $error */
+                printf(
+                    "\t%s: %s%s",
+                    $error->getErrorCode()->getErrorCode(),
+                    $error->getMessage(),
+                    PHP_EOL
+                );
+            }
+        } catch (ApiException $apiException) {
+            printf(
+                "ApiException was thrown with message '%s'.%s",
+                $apiException->getMessage(),
+                PHP_EOL
+            );
+        }
+    }
+
+    /**
+     * Runs the example.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     */
+    public static function runExample(GoogleAdsClient $googleAdsClient, int $customerId)
```

Comment:
@wihl had an excellent suggestion: if we make the manager customer ID optional, then this would be even more useful. If the user does not specify a manager customer ID, you could then call `listAccessibleCustomers` to get a list of managers accessible via the credentials, and then fetch and print the tree for each one of those managers.

Reply:
Let me confirm a bit more so I'm sure I understand correctly.
So, based on this suggestion we will:
- If no manager customer ID specified, call listAccessibleCustomers
- For each customer ID, call GoogleAdsService.Search() to check which ones are managers
- For all the managers, do what we originally do in this example -- getting all customer accounts under them.
?

Reply:
And what happen if we have more than one root manager customer IDs?
Do we need to check if the accessible customers are actually child account of the other in the listAccessibleCustomers?

I find it's a bit complicated for the result printing process if we go down this path.

Reply:
Thanks for clarification. But that's actually a subset of what is possible.
In case we want to expand this example to use listaccessiblecustomers, we probably need to cover all possible cases.

In particular, what should we do when there are many customers accessible by the authenticated account, especially when some of them are child accounts of the others? Should we need to implement an algorithm to check the parent-child relationship? (That would further lengthen this example though) or should we just merely print the results for each accessible customers?

Reply:
Alright. Done.
Note that there are some places where I modified the variable to be just a customer ID because accessible customers can be non-manager ones, and I don't want to further complicate this example by trying to figure out whether each accessible customer is a manager or just a client.
This example in fact also works with non-manager customers too.

For the optional argument of the example, I retain it as manager customer ID though because it's more useful to use this example with a manager customer ID.